src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -159,161 +145,117 @@ public Response getClusterTables(@PathParam("cluster_name") String clusterName)
-        return Response.notModified()
-            .entity(
-                String.format(
-                    "Topology hasn't changed in cluster %s. No update was performed.",
-                    existingCluster.get().getName()))
-            .build();
-      } else {
-        LOG.info(
-            "Node list of cluster {} has been updated with the current topology.",
-            existingCluster.get().getName());
-        return Response.ok()
-            .entity(String.format("Updated cluster %s node list", existingCluster.get().getName()))
-            .build();
+      Cluster cluster = findClusterWithSeedHost(seedHost.get());
+      if (null == cluster) {
+        String msg = String.format("failed to find cluster %s with seed host %s", clusterName.or(""), seedHost.get());
+        return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
       }
-    } else {
-      LOG.info("creating new cluster based on given seed host: {}", newCluster.getName());
-      context.storage.addCluster(newCluster);
-
-      if (context.config.hasAutoSchedulingEnabled()) {
-        try {
-          clusterRepairScheduler.scheduleRepairs(newCluster);
-        } catch (ReaperException e) {
-          LOG.error("failed to automatically schedule repairs", e);
-          return Response.status(400)
-              .entity(
-                  "failed to automatically schedule repairs for cluster with seed host \""
-                  + seedHost.get()
-                  + "\". Exception was: "
-                  + e.getMessage())
-              .build();
+      if (clusterName.isPresent() && !cluster.getName().equals(clusterName.get())) {
+
+        String msg = String.format(
+            "POST/PUT on cluster resource %s called with seedHost %s belonging to different cluster %s",
+            clusterName.get(),
+            seedHost.get(),
+            cluster.getName());
+
+        LOG.info(msg);
+        return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
+      }
+      Optional<Cluster> existingCluster = context.storage.getCluster(cluster.getName());
+      URI location = uriInfo.getBaseUriBuilder().path("cluster").path(cluster.getName()).build();
+      if (existingCluster.isPresent()) {
+        LOG.debug("Attempting updating nodelist for cluster {}", existingCluster.get().getName());
+
+        cluster = updateClusterSeeds(existingCluster.get(), seedHost.get());
+        if (cluster.getSeedHosts().equals(existingCluster.get().getSeedHosts())) {
+          LOG.debug("Nodelist of cluster {} is already up to date.", existingCluster.get().getName());
+          return Response.noContent().location(location).build();
+        } else {
+          LOG.info("Nodelist of cluster {} updated", existingCluster.get().getName());
+          return Response.ok().location(location).build();
+        }
+      } else {
+        LOG.info("creating new cluster based on given seed host: {}", cluster.getName());
```

Comment:
Nit pick: Check to see if all the capitalisation in the log messages is consistent.